src/app/_components/ClientPagination.tsx

```diff
@@ -0,0 +1,66 @@
+'use client'
+
+import clsx from 'clsx'
+
+type PaginationProps = {
+  currentPage: number
+  total: number
+  size: number
+  setCurrentPage: (page: number) => void
+}
+
+export function ClientPagination({
+  currentPage,
+  setCurrentPage,
+  total,
+  size,
+}: PaginationProps) {
+  const pages = Math.ceil(total / size)
```

Comment:
Should be memoize this to avoid unnecessary recalculations on each render?

```suggestion
  const pages = useMemo(() => Math.ceil(total / size), [total, size])
```